app/models/concerns/taggable.rb

```diff
@@ -9,6 +9,14 @@ module Taggable
     # Assumes that base model implements a 'tags' array column. Being this
     # opinionated keeps the codebase simpler.
 
+    def with_no_tags
+      where("tags = '{}'")
+    end
+
+    def with_any_tags
+      where.not("tags = '{}'")
```

Comment:
Might be worth checking the performance characteristics of these using `EXPLAIN` - does it use the index?